testutil/blobfactory/payforblob_factory.go

```diff
@@ -310,13 +311,11 @@ func ManyBlobs(t *testing.T, namespaces []appns.Namespace, sizes []int) []*tmpro
 
 func NestedBlobs(t *testing.T, namespaces []appns.Namespace, sizes [][]int) [][]*tmproto.Blob {
 	blobs := make([][]*tmproto.Blob, len(sizes))
-	counter := 0
 	for i, set := range sizes {
 		for _, size := range set {
-			blob, err := blobtypes.NewBlob(namespaces[counter], tmrand.Bytes(size), appconsts.ShareVersionZero)
+			blob, err := blobtypes.NewBlob(namespaces[i], tmrand.Bytes(size), appconsts.ShareVersionZero)
 			require.NoError(t, err)
 			blobs[i] = append(blobs[i], blob)
-			counter++
```

Comment:
[blocking question]
wait, why was this done? won't this result in not using the caller specified namespaces?

Reply:
I was trying to simplify this code but on second look, the change wasn't needed and I don't think it's correct. The fact that no tests broke after this change makes me think that nested blob test data isn't really verified but I'll revert.

Reply:
resolved by ae01b98416c38a218b0de5d37cebb8c553a6001c

Reply:
> The fact that no tests broke after this change makes me think that nested blob test data isn't really verified

this just overrides the callers namespace, so unless we're checking for the caller's namespace, I wouldn't expect it to be detected